d3d12/src/debug.rs

```diff
@@ -40,4 +43,14 @@ impl Debug {
     pub fn enable_layer(&self) {
         unsafe { self.EnableDebugLayer() }
     }
+
+    pub fn enable_gpu_based_validation(&self) -> bool {
+        let (ptr, hr) = unsafe { self.cast::<d3d12sdklayers::ID3D12Debug1>() };
+        if hr == S_OK {
+            unsafe { ptr.SetEnableGPUBasedValidation(TRUE) };
+            true
+        } else {
+            false
+        }
```

Comment:
This really should be using weak_com_inheritance_chain for `Debug` for this instead of manually casting.

Reply:
Ah, I was confused for a second, because I thought you were referring to `std::fmt::Debug`. 😆 Right-o! 

Gave it a crack with 2ba67bf59. How's it look?

Reply:
So apparently the inheritance is all messed up. It's:

```
3 -> 0
4 -> 3
5 -> 4
6 -> 5
```

Which... bleh. So I rescind my comment - lets query interface our way to victory.

Reply:
Reverted!